test/Core.Test/Services/Fido2AuthenticatorGetAssertionTests.cs

```diff
@@ -148,38 +155,61 @@ public async Task GetAssertionAsync_AsksForDiscoverableCredentials_ParamsDoesNot
 
         [Fact]
         // Spec: Prompt the user to select a public key credential source `selectedCredential` from `credentialOptions`.
-        //       If requireUserVerification is true, the authorization gesture MUST include user verification.
+        //       If UserVerificationPreference is Required, the authorization gesture MUST include user verification.
         public async Task GetAssertionAsync_RequestsUserVerification_ParamsRequireUserVerification()
         {
             // Arrange
-            _params.RequireUserVerification = true;
+            _params.UserVerificationPreference = Fido2UserVerificationPreference.Required;
             _userInterface.PickCredentialAsync(Arg.Any<Fido2GetAssertionUserInterfaceCredential[]>()).Returns((_ciphers[0].Id, true));
 
             // Act
             await _sutProvider.Sut.GetAssertionAsync(_params, _userInterface);
 
             // Assert
             await _userInterface.Received().PickCredentialAsync(Arg.Is<Fido2GetAssertionUserInterfaceCredential[]>(
-                (credentials) => credentials.All((c) => c.RequireUserVerification == true)
+                (credentials) => credentials.All((c) => c.UserVerificationPreference == Fido2UserVerificationPreference.Required)
+            ));
+        }
+
+        [Fact]
+        // Spec: Prompt the user to select a public key credential source `selectedCredential` from `credentialOptions`.
+        //       If UserVerificationPreference is Preferred and MP reprompt is on then the authorization gesture MUST include user verification.
+        //       If MP reprompt is off then the authorization gestue MAY include user verification
+        public async Task GetAssertionAsync_RequestsPreferredUserVerification_ParamsPreferUserVerification()
+        {
+            // Arrange
+            _params.UserVerificationPreference = Fido2UserVerificationPreference.Preferred;
+            _userInterface.PickCredentialAsync(Arg.Any<Fido2GetAssertionUserInterfaceCredential[]>()).Returns((_ciphers[0].Id, true));
+
+            // Act
+            await _sutProvider.Sut.GetAssertionAsync(_params, _userInterface);
+
+            // Assert
+            await _userInterface.Received().PickCredentialAsync(Arg.Is<Fido2GetAssertionUserInterfaceCredential[]>(
+                (credentials) => credentials.Any((c) => _ciphers.First(cip => cip.Id == c.CipherId).Reprompt == CipherRepromptType.None && c.UserVerificationPreference == Fido2UserVerificationPreference.Preferred)
+            ));
+
+            await _userInterface.Received().PickCredentialAsync(Arg.Is<Fido2GetAssertionUserInterfaceCredential[]>(
+                (credentials) => credentials.Any((c) => _ciphers.First(cip => cip.Id == c.CipherId).Reprompt != CipherRepromptType.None && c.UserVerificationPreference == Fido2UserVerificationPreference.Required)
             ));
```

Comment:
❓ question: why do we have two checks here?

Reply:
Ohh I didn't consider that case! But doesn't this test check that `PickCredentialAsync` has been called twice? `           GetAssertionAsync` should only call it one?

Reply:
Ohh right, I was getting confused because I saw `First(...)` and thought it was getting the first element in the array and checking it against Reprompt and UV, but the check are mutually exclusive so I couldn't figure out how one of those didn't fail. But of course that's not what it's doing, `First(...)` is actually searching for the first element that matches the statement